### DISCUSSION TITLE
JWT authentication.

### DISCUSSION THREAD

**Discussion by ramonamezquita** [grafana/grafana-infinity-datasource#368] [Q&A]
Hello! I'm trying to gather data from an API that uses JWT authentication. Is it possible for Grafana to login to the API, retrieve the token (which is in the response body of the login endpoint ), and use it for further requests?
It will also be required that once the expiration time for the token ends, Grafana repeats the login process to obtain a new one.

The login/token endpoint requieres two parameters to be specified, ``username`` and ``password``; however, it seems infinity uses a different set of parameters.

Furthermore, this API is currently under development, so if needed, I can modify the login parameters.

**Comment by tmur11** [comment]
@ramonamezquita did you ever figure this out?

**Reply by ramonamezquita** [reply]
I have not used any of this in a while, but I remember I stopped trying JWT and changed to a basic authentication method 😔.

**Reply by tmur11** [reply]
Okay bummer! I unfortunately don't control this endpoint API and it requires basic authentication initally to obtain a jwt.
